kodi_game_scripting/process_game_addons.py

```diff
@@ -136,8 +136,9 @@ def process(self):
         for addon in self._addons.values():
             print(" Processing addon: {}".format(addon.name))
             addon.process_addon_files()
-            print(" Processing addon description: {}".format(addon.name))
-            addon.process_description_files()
+            if self._args.push_branch:
```

Comment:
If you want to add a new core, you won't be able to compile if there's no addon description file. In that case the branch doesn't really matter.
Therefore I think it's better to always generate descriptions and maybe default to 'master' in that jinja2 template.